### PR TITLE
Rust-Nightly autom. CI-Update

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**
I tried out things here:
https://github.com/geo-engine/ci-tests/blob/main/.github/workflows/cargo.yml

That led to:
https://github.com/geo-engine/ci-tests/pull/1

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Rust-Nightly autom. CI-Update](https://trello.com/c/gsOljzvV/117-rust-nightly-autom-ci-update)

File: .github/workflows/fix-nightly.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          commit-message: Update Rust Nightly Version
+          title: Update Rust Nightly Version
+          body: |
+            - Fix a new Rust Nightly version
+
+            Auto-generated PR
+          branch: update-rust-nightly-version
```
Comment (michaelmattig) [inline .github/workflows/fix-nightly.yml:26]: does this work when last weeks PR ist net yet merged or ist there a name conflict then?

File: .github/workflows/fix-nightly.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set New Rust Nightly Version
+        run: |
+          sed -i -r "s:(toolchain\: nightly).+:\1-`date --date=yesterday +%Y-%m-%d`:g" .github/workflows/ci.yml
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          commit-message: Update Rust Nightly Version
```
Comment (michaelmattig) [inline .github/workflows/fix-nightly.yml:20]: who is the author of the commit? github actions? Will soon be a major contributor of this project then

**Comment by michaelmattig** [comment]
so the checks have failed, how would we now proceed with the PR? Who fixes the new clippy errors? Will they be fixed in this PR or in another one and then we merge the master into this one?

**Comment by ChristianBeilschmidt** [comment]
Depends on https://github.com/geo-engine/geoengine/pull/94

File: .github/workflows/fix-nightly.yml (hunk 2 above)
Comment (ChristianBeilschmidt) [reply .github/workflows/fix-nightly.yml:20]: Yes, but I guess this is okay.

File: .github/workflows/fix-nightly.yml (hunk 1 above)
Comment (ChristianBeilschmidt) [reply .github/workflows/fix-nightly.yml:26]: From the docs:

> The default behaviour of the action is to create a pull request that will be continually updated with new changes until it is merged or closed. Changes are committed and pushed to a fixed-name branch, the name of which can be configured with the branch input. Any subsequent changes will be committed to the same branch and reflected in the open pull request.

So it should just update the branch then.

**Comment by michaelmattig** [comment]
> 
> 
> Depends on #94

And in the future? Will we always create a new clippy-fix pr, review ist separately und then merge it into the fix-nightly-pr? Or do we fix it in the PR? Who will be responsible to do it?

**Comment by ChristianBeilschmidt** [comment]
> > Depends on #94
> 
> And in the future? Will we always create a new clippy-fix pr, review it separately und then merge it into the fix-nightly-pr? Or do we fix it in the PR? Who will be responsible to do it?

We will fix this error in the PR-branch directly if it occurs. Wrt. the responsibility, it is the same for the dependency updates. We have to see how this works out and otherwise discuss it at some point.

**Pull request merged**